Tests/GRDBTests/RecordMinimalPrimaryKeyRowIDTests.swift

```diff
@@ -380,6 +381,7 @@ class RecordMinimalPrimaryKeyRowIDTests : GRDBTestCase {
             
             let fetchedRecord = try MinimalRowID.filter(key: ["id": record.id]).fetchOne(db)!
             XCTAssertTrue(fetchedRecord.id == record.id)
+            XCTAssertEqual(lastSQLQuery, "SELECT * FROM \"minimalRowIDs\" WHERE (\"id\" = \(record.id!)) LIMIT 1")
```

Comment:
I'm very happy you added this test.

I'm somewhat sorry, because this technique for fetching one record by primary or unique key wasn't listed in my "exhaustive" list in #513:

```swift
let request = Player.filter(key: 123) // or Player.filter(key: ["email": "me@example.com"])
let player = try dbQueue.read { db in try request.fetchOne(db) }
```

For the context, this alternative way to perform `fetchOne(_:key:)` is useful because such requests can be observed so that the application is notified of db changes:

```swift
let observation = ValueObservation(trackingOne: Player.filter(key: 123))
try observation.start(in: dbQueue) { player: Player?
    print("Player has changed: \(player)")
} 
```

**What about trying to remove `LIMIT 1` here as well?**

One way to achieve this goal is to add a flag in SQLSelectQuery. When this flag is set, we could ignore the `limitOne` hint. And maybe we'd no longer need the `primaryKeyOrUnique` hint after that?

This would be a safe technique because all request derivation methods can only *increase* the amount of filtering, never relax it:

```swift
// Player 123, along with its team, but only if it is active:
//
// SELECT player.*, team.*
// FROM player
// JOIN team ON team.id = player.teamId
// WHERE player.id = 123 AND player.isActive = 1
let request = Player
    .filter(key: 123)
    .filter(Column("isActive") == true)
    .including(required: Player.team)
```

We'd need to make sure that all SQLSelectQuery derivation methods transfer this flag from one query to another. But after a quick glance, it looks like all those methods are already information-preserving.

Reply:
NB: this change is not required for this pull request to enter review. We can proceed without it if you prefer. But if you agree with the suggestion, and if you have a little more time, this change would be appreciated.

Reply:
Hint: look for `filter<PrimaryKeyType: DatabaseValueConvertible>(key:` and `filter(key:`.

Reply:
Good suggestion @groue, and thanks for the hints 😄 

Definitely agree it'd be good to remove LIMIT 1 from the SQL there. I can look into it.

Reply:
Cool, thank you!

Reply:
Ok I think I've got that implemented as you described it. Tests are updated to exclude LIMIT 1 from the SQL in that scenario.